### DISCUSSION TITLE
[Bug] How to use Russian data for fine-tuning？

### DISCUSSION THREAD

**Discussion by YangGao1001** [idiap/coqui-ai-TTS#429] [Q&A] [answered]
### Describe the bug

I am currently using  recipes/ljspeech/xtts_v2/train_gpt_xtts.py  and 98 hours of Russian audio data to fine tune coqui/XTTS-v2/mol.Pth. The training data format is ljSpeech, but after fine-tuning, the model's error rate has actually increased. May I ask if I am using the wrong training code or if the training data is insufficient? Or what other reasons could it be?

### To Reproduce

```python
import os

from trainer import Trainer, TrainerArgs

from TTS.config.shared_configs import BaseDatasetConfig
from TTS.tts.datasets import load_tts_samples
from TTS.tts.layers.xtts.trainer.gpt_trainer import GPTArgs, GPTTrainer, GPTTrainerConfig
from TTS.tts.models.xtts import XttsAudioConfig
from TTS.utils.manage import ModelManager

os.environ["CUDA_VISIBLE_DEVICES"] = "0"

# Logging parameters
RUN_NAME = "XTTS-v2-train-98h"
PROJECT_NAME = "XTTS_trainer"
DASHBOARD_LOGGER = "tensorboard"
LOGGER_URI = None

# Set here the path that the checkpoints will be saved. Default: ./run/training/
OUT_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), "run", "training-98h")

# Training Parameters
OPTIMIZER_WD_ONLY_ON_WEIGHTS = True  # for multi-gpu training please make it False
START_WITH_EVAL = True  # if True it will star with evaluation
BATCH_SIZE = 3  # set here the batch size
GRAD_ACUMM_STEPS = 84  # set here the grad accumulation steps
# Note: we recommend that BATCH_SIZE * GRAD_ACUMM_STEPS need to be at least 252 for more efficient training. You can increase/decrease BATCH_SIZE but then set GRAD_ACUMM_STEPS accordingly.

# Define here the dataset that you want to use for the fine-tuning on.
config_dataset = BaseDatasetConfig(
    formatter="ljspeech",
    dataset_name="RuLS_Librispeech",
    path="/root/autodl-tmp/DividedDataSet/RuLS_Librispeech",
    meta_file_train="/root/autodl-tmp/DividedDataSet/RuLS_Librispeech/metadata.txt",
    language="ru",
)

# Add here the configs of the datasets
DATASETS_CONFIG_LIST = [config_dataset]


CHECKPOINTS_OUT_PATH = "/root/autodl-tmp/coqui-ai-TTS-dev/coqui/XTTS-v2"
DVAE_CHECKPOINT = os.path.join(CHECKPOINTS_OUT_PATH, "dvae.pth")
MEL_NORM_FILE = os.path.join(CHECKPOINTS_OUT_PATH, "mel_stats.pth")
TOKENIZER_FILE = os.path.join(CHECKPOINTS_OUT_PATH, "vocab.json")  # vocab.json file
XTTS_CHECKPOINT = os.path.join(CHECKPOINTS_OUT_PATH, "model.pth")  # model.pth file


# Training sentences generations
SPEAKER_REFERENCE = [
    "/root/autodl-tmp/coqui-ai-TTS-dev/audio/LiBai-JingYesi-ru/6.wav",
    "/root/autodl-tmp/coqui-ai-TTS-dev/audio/LiBai-JingYesi-ru/7.wav",
    "/root/autodl-tmp/coqui-ai-TTS-dev/audio/LiBai-JingYesi-ru/8.wav",
    "/root/autodl-tmp/coqui-ai-TTS-dev/audio/LiBai-JingYesi-ru/9.wav",
    "/root/autodl-tmp/coqui-ai-TTS-dev/audio/LiBai-JingYesi-ru/10.wav"    # speaker reference to be used in training test sentences
]
LANGUAGE = config_dataset.language


def main():
    # init args and config
    model_args = GPTArgs(
        max_conditioning_length=132300,  # 6 secs
        min_conditioning_length=66150,  # 3 secs
        debug_loading_failures=False,
        max_wav_length=255995,  # ~11.6 seconds
        max_text_length=200,
        mel_norm_file=MEL_NORM_FILE,
        dvae_checkpoint=DVAE_CHECKPOINT,
        xtts_checkpoint=XTTS_CHECKPOINT,  # checkpoint path of the model that you want to fine-tune
        tokenizer_file=TOKENIZER_FILE,
        gpt_num_audio_tokens=1026,
        gpt_start_audio_token=1024,
        gpt_stop_audio_token=1025,
        gpt_use_masking_gt_prompt_approach=True,
        gpt_use_perceiver_resampler=True,
    )
    # define audio config
    audio_config = XttsAudioConfig(sample_rate=22050, dvae_sample_rate=22050, output_sample_rate=24000)
    # training parameters config
    config = GPTTrainerConfig(
        output_path=OUT_PATH,
        model_args=model_args,
        run_name=RUN_NAME,
        project_name=PROJECT_NAME,
        run_description="""
            GPT XTTS training
            """,
        dashboard_logger=DASHBOARD_LOGGER,
        logger_uri=LOGGER_URI,
        audio=audio_config,
        batch_size=BATCH_SIZE,
        batch_group_size=48,
        eval_batch_size=BATCH_SIZE,
        num_loader_workers=8,
        eval_split_max_size=256,
        eval_split_size=0.1,  # gy
        print_step=50,
        plot_step=100,
        log_model_step=1000,
        save_step=13000,
        save_n_checkpoints=3,
        save_checkpoints=True,
        # target_loss="loss",
        print_eval=False,
        # Optimizer values like tortoise, pytorch implementation with modifications to not apply WD to non-weight parameters.
        optimizer="AdamW",
        optimizer_wd_only_on_weights=OPTIMIZER_WD_ONLY_ON_WEIGHTS,
        optimizer_params={"betas": [0.9, 0.96], "eps": 1e-8, "weight_decay": 1e-2},
        lr=5e-06,  # learning rate
        lr_scheduler="MultiStepLR",
        # it was adjusted accordly for the new step scheme
        lr_scheduler_params={"milestones": [50000 * 18, 150000 * 18, 300000 * 18], "gamma": 0.5, "last_epoch": -1},
        test_sentences=[
            {
                "text": "Когда я вернулся домой, два моих брата уже разделили наследство.",
                "speaker_wav": SPEAKER_REFERENCE,
                "language": LANGUAGE,
            },
            {
                "text": "Потом я вернулся и стал учиться.",
                "speaker_wav": SPEAKER_REFERENCE,
                "language": LANGUAGE,
            },
        ],
    )

    # init the model from config
    model = GPTTrainer.init_from_config(config)

    # load training samples
    train_samples, eval_samples = load_tts_samples(
        DATASETS_CONFIG_LIST,
        eval_split=True,
        eval_split_max_size=config.eval_split_max_size,
        eval_split_size=config.eval_split_size,
    )
    
    # 提取验证集的音频路径
    eval_audio_paths = [sample["audio_file"] for sample in eval_samples]

    # 保存为 txt 文件
    output_txt_path = "eval_audio_paths.txt"
    with open(output_txt_path, "w", encoding="utf-8") as f:
        for path in eval_audio_paths:
            f.write(path + "\n")


    # init the trainer and 🚀
    trainer = Trainer(
        TrainerArgs(
            restore_path=None,  # xtts checkpoint is restored via xtts_checkpoint key so no need of restore it using Trainer restore_path parameter
            skip_train_epoch=False,
            start_with_eval=START_WITH_EVAL,
            grad_accum_steps=GRAD_ACUMM_STEPS,
        ),
        config,
        output_path=OUT_PATH,
        model=model,
        train_samples=train_samples,
        eval_samples=eval_samples,
    )
    trainer.fit()


if __name__ == "__main__":
    main()
```

### Logs

```shell

```

### Environment

```shell
Linux
```

**Comment by eginhard** [comment, marked as answer]
As I wrote on Discord, you'll need to listen to the outputs to see if training is successful. The loss is not necessarily a good indicator.

**Reply by YangGao1001** [reply]
ok，thank you!
